packages/core/src/types.ts

```diff
@@ -152,15 +153,17 @@ export function indexSearchParameterBundle(bundle: Bundle<SearchParameter>): voi
   }
 }
 
-/**
- * Indexes a SearchParameter resource for fast lookup.
- * Indexes by SearchParameter.code, which is the query string parameter name.
- * @param searchParam - The SearchParameter resource.
- * @see {@link IndexedStructureDefinition} for more details on indexed StructureDefinitions.
- */
-export function indexSearchParameter(searchParam: SearchParameter): void {
-  for (const resourceType of searchParam.base ?? []) {
-    let typeSchema = globalSchema.types[resourceType];
+export function indexDefaultSearchParameters(bundle: StructureDefinition[] | Bundle): void {
+  const sds = Array.isArray(bundle) ? bundle : (bundle.entry?.map((e) => e.resource as StructureDefinition) ?? []);
+  for (const sd of sds) {
+    if(sd.resourceType === 'StructureDefinition'){
+      getOrInitTypeSchema(sd.type);
+    }
+  }
+}
+
+export function getOrInitTypeSchema(resourceType: string): TypeInfo {
```

Comment:
Does this function need to be exported?
```suggestion
function getOrInitTypeSchema(resourceType: string): TypeInfo {
```